code/modules/urist/modules/wfc/core/turf.dm

```diff
@@ -0,0 +1,40 @@
+/turf
+	var/wfc_overwritable = FALSE
+
+	// list of refs to objects to delete on overwrite
+	var/list/associated_wfc_overwritables = null
+
+
+/turf/procgen
+	icon = 'icons/urist/turf/uristturf.dmi'
+	icon_state = "tramfloor"
+
+	wfc_overwritable = TRUE
+
+
+/turf/procgen/wfc
+
+
+/proc/WfcChangeTurf(var/turf/trg, var/new_type)
+	if(isnull(new_type))
+		return
+
+	if(!ispath(new_type))
+		return
+
+	if(!ispath(new_type, /turf))
+		return
+
+	if(!isnull(trg?.associated_wfc_overwritables))
+		for(var/overwritten in trg.associated_wfc_overwritables)
+			qdel_from_weakref(overwritten)
+
+	trg.icon_state = initial(trg.icon_state)
```

Comment:
Seems odd we're accounting for a null `trg` above, but not here?

Reply:
Good spot, oversight.